### DISCUSSION TITLE
Why was AsyncValue.previous removed?

### DISCUSSION THREAD

**Discussion by NickBalnaves** [rrousselGit/riverpod#905] [Q&A] [answered]
Is there another way to get AsyncValue.previous in a FutureProvider which was removed In this commit https://github.com/rrousselGit/river_pod/commit/cb9ce91dea5a652a0a0f7f753ebba682a1b829f5?
I have a FutureProvider that occasionally gets refreshed with ref.refresh and need its previous state before the refresh changed it into its loading state. 

```dart
final exampleProvider = FutureProvider.autoDispose<Example>(
  (ref) async {
     ...
  },
);

final otherExampleProvider = Provider.autoDispose<OtherExample>(
  (ref) {
    return ref.watch(exampleProvider).when(
      data: ...,
      error: ...,
      loading: (data) { // data was available before the commit
        final previous = data?.asData?.value; 
        ...
      },
    );
  },
);

ref.refresh(exampleProvider);
```

**Comment by TimWhiting** [comment]
The reason I believe was that it interacted badly with custom uses of AsyncValue in StateNotifiers. Remi is trying to think of a better solution to this problem.

**Comment by rrousselGit** [comment, marked as answer]
As Tim mentioned, this solution didn't scale well with provider variants. The syntax also wasn't ideal

And I strongly believe that there is a much better solution lurking around. After the devtool, that's likely the next problem I will tackle, so you shouldn't have to wait too long

**Comment by tomoyuki28jp** [comment]
Is there any progress?

**Comment by jamesdixon** [comment]
Also curious to see if there's a new solution for this. Like @NickBalnaves, I have a `FutureProvider` where I would prefer to display the last known value until it's been refreshed. My users are offline quite a bit so being able to default to the previous known value would be 🙌

**Reply by NickBalnaves** [reply]
@jamesdixon the new 2.0.0-dev.0 release does this
